includes/class-wc-payments-woopay-direct-checkout.php

```diff
@@ -20,7 +20,7 @@ class WC_Payments_WooPay_Direct_Checkout {
 	 * @return void
 	 */
 	public function init() {
-		add_action( 'wp_enqueue_scripts', [ $this, 'scripts' ] );
+		add_action( 'wp_footer', [ $this, 'scripts' ] );
```

Comment:
This is because the mini cart widget can be added to as late as in the footer section of the template.

Reply:
The issue is not about moving the scripts to the footer, but it's in detecting if `did_action( 'woocommerce_blocks_cart_enqueue_data' )` had already been fired, as the mini cart can get added to a footer column for example.

We use `woocommerce_blocks_cart_enqueue_data` to tell if the blocks mini cart is present on the page.

Reply:
Ah, I see. That makes sense 👍